hermes/src/network/pythnet.rs

```diff
@@ -317,6 +331,117 @@ pub async fn spawn(opts: RunOptions, state: Arc<State>) -> Result<()> {
         })
     };
 
-    let _ = tokio::join!(task_listener, task_guadian_watcher);
+
+    let task_price_feeds_metadata_updater = {
+        let price_feeds_state = state.clone();
+        tokio::spawn(async move {
+            while !crate::SHOULD_EXIT.load(Ordering::Acquire) {
+                if let Err(e) = fetch_and_store_price_feeds_metadata(
+                    price_feeds_state.as_ref(),
+                    &opts.pythnet.mapping_addr,
+                    &rpc_client,
+                )
+                .await
+                {
+                    tracing::error!("Error in fetching and storing price feeds metadata: {}", e);
+                }
+                // This loop with a sleep interval of 1 second allows the task to check for an exit signal at a
+                // fine-grained interval. Instead of sleeping directly for the entire `price_feeds_update_interval`,
+                // which could delay the response to an exit signal, this approach ensures the task can exit promptly
+                // if `crate::SHOULD_EXIT` is set, enhancing the responsiveness of the service to shutdown requests.
+                for _ in 0..DEFAULT_PRICE_FEEDS_CACHE_UPDATE_INTERVAL {
+                    if crate::SHOULD_EXIT.load(Ordering::Acquire) {
+                        break;
+                    }
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                }
```

Comment:
We should probably replace the exit variable with a watch channel at some point.